app/src/main/java/com/gathering/android/event/myevent/MyEventScreen.kt

```diff
@@ -43,74 +52,110 @@ class MyEventScreen : Fragment(), MyEventNavigator {
     ): View {
-        binding = ScreenMyEventBinding.inflate(layoutInflater)
-        return binding.root
+        return if (!isComposeEnabled) {
+            binding = ScreenMyEventBinding.inflate(layoutInflater)
+            return binding.root
+        } else {
+            ComposeView(requireContext()).apply {
+                setContent {
+                    GatheringTheme {
+                        Surface(
+                            modifier = Modifier.wrapContentSize(),
+                            color = MaterialTheme.colorScheme.background
+                        ) {
+                            val state = viewModel.uiState.collectAsState()
+
+                            EventList(
+                                showEditIcon = true,
+                                showFavoriteIcon = false,
+                                events = state.value.myEvents,
+                                onItemClick = {},
+                                onEditClick = { viewModel.onEditEventClicked(it) },
+                                onFavClick = {},
+                                isLoading = state.value.showProgress,
+                                isNoData = state.value.showNoData,
+                                onFabClick = viewModel::onFabButtonClicked,
+                                onDeleteClick = { viewModel.onSwipedToDelete(it) },
+                                onUndoDeleteEvent = {viewModel.onUndoDeleteEvent(it)},
+                                swipeEnabled = true
+                            )
+                        }
+                    }
+                }
+            }
+        }
     }
 
     override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
         super.onViewCreated(view, savedInstanceState)
-        val linearLayoutManager =
-            LinearLayoutManager(requireContext(), LinearLayoutManager.VERTICAL, false)
-        binding.rvEvent.layoutManager = linearLayoutManager
-
-        swipeGesture = object : SwipeGesture(requireContext()) {
-            override fun onSwiped(viewHolder: RecyclerView.ViewHolder, direction: Int) {
-                val position = viewHolder.absoluteAdapterPosition
-                val event = adapter.getEventAtPosition(position)
-                event?.let {
-                    if (direction == ItemTouchHelper.LEFT) {
-                        viewModel.onSwipedToDelete(it)
-                        // Show a snack bar with an "Undo" option for the delete action
-                        val snackBar = Snackbar.make(
-                            binding.root, EVENT_DELETED, Snackbar.LENGTH_LONG
-                        )
-                        snackBar.setAction(UNDO) {
-                            viewModel.onUndoDeleteEvent()
+        if (isComposeEnabled) {
+            viewModel.onViewCreated(this)
+            return
+        } else {
+            val linearLayoutManager =
```

Comment:
its a very long block of code. would you please extract the else block contents into a method. could be called bindNonComposeView()

Reply:
how is this for the name of the method: "setupRecyclerViewAndInteractions" ?